src/element_interfaces/debug_data.rs

```diff
@@ -0,0 +1,253 @@
+use crate::callsystem::*;
+use crate::codata::*;
+use crate::neutronerror::NeutronError::*;
+use crate::neutronerror::*;
+use core::mem::transmute;
+use neutron_common::RecoverableError;
+use std::str;
+/*
+## Debug Data Injector ElementAPI
+
+ID: 0x8000_0001
+
+This API is used in a testing environment to inject data into a test instance of the Neutron stack
+This file also contains some data structures that simplifies construction of this data
+
+Data structures:
+
+* mock_input_stack          - An initial input stack state that the testing smart contract can load
+* expected_output_stack     - An expected state for the output stack that the testing smart contract can asserted against,
+                              along with debug information to make a failed assertion more informative than "bytes in stacks didn't match"
+
+functions:
+
+* [0] Available()           - Check if API is available in current instance (standard function)
+* [1] PushInputStack()      - Pushes the provided mock input stack to the current instance
+* [2] AssertOutputStack()   - Asserts the provided expected output stack against the current instance
+
+*/
+
+// IDs >= 0x8000_0000 are meant for "internal" usage within the project, e.g. for unit/integration testing.
+pub const DEBUG_DATA_FEATURE: u32 = 0x8000_0001;
```

Comment:
Change this to 0x4000_0001 to avoid potential conflicts with blockchain provided element IDs which start at 0x8000_0000

Reply:
Small question on that: 

I read this code comment as >= 0x8000_0000 being "private" as in internal to the project, which seems to be a good fit for a testing function. Am I misunderstanding this?

    pub fn call(&self, codata: &mut CoData, element: u32, function: u32) -> Result<ElementResult, NeutronError>{
        let function = function & (!0x8000_0000); //public calls can not set the top bit, which is reserved for private functions
        self.private_call(codata, element, function)
    }

Reply:
Nevermind, I'll change it.